tcms/testcases/static/testcases/js/search.js

```diff
@@ -118,6 +118,34 @@ $(document).ready(function() {
 
     hookIntoPagination('#resultsTable', table);
 
+    $("#check_all").click(function () {
+        if ($("input:checkbox").prop("checked")) {
+            $("input:checkbox[name='row-check']").prop("checked", true);
+            table.rows().select()
+        } else {
+            $("input:checkbox[name='row-check']").prop("checked", false);
+            table.rows().deselect()
+        }
+    });
+
+    table.on('select', function (e, dt, type, indexes) {
+        if (type === 'row') {
+            let total_check_boxes = $("input:checkbox[name='row-check']").length;
+            let total_checked_boxes = $("input:checkbox[name='row-check']:checked").length;
+            if ((total_check_boxes === total_checked_boxes)) {
+                $("input:checkbox").prop("checked", true)
+            }
+        }
+    });
+
+    table.on('deselect', function (e, dt, type, indexes) {
+        if (type === 'row') {
```

Comment:
out of curiosity, what other values could `type` have?

Reply:
API docs mention that it could be a column or a cell https://datatables.net/reference/event/deselect